src/modules/DASDController.rb

```diff
@@ -532,215 +407,20 @@ def DeactivateDisk(channel, diag)
-              # integers,
-              # eg. Formatting /dev/dasda: cylinder 123 of 12334 done
-              _("Formatting %1: cylinder %2 of %3 done"),
-              Ops.get(disks, k, ""),
-              Ops.get(done, k, 0),
-              Ops.get(cylinders, k, 0)
-            )
-          )
-          UI.ChangeWidget(
-            Id(index),
-            :Value,
-            Ops.divide(
-              Ops.multiply(100, Ops.get(done, k, 0)),
-              Ops.get(cylinders, k, 1)
-            )
-          )
-          UI.ChangeWidget(Id(index), :Enabled, true)
-          index = Ops.add(index, 1)
-        end
-        while Ops.less_than(index, par)
-          UI.ChangeWidget(Id(index), :Label, "")
-          UI.ChangeWidget(Id(index), :Value, 0)
-          UI.ChangeWidget(Id(index), :Enabled, false)
-          index = Ops.add(index, 1)
-        end
-      end
-      UI.CloseDialog
-      iret = Convert.to_integer(SCR.Read(path(".process.status"), process_id))
-      if iret != 0
-        # error report, %1 is exit code of the command (integer), %2 output of command
-        Report.Error(
-          Builtins.sformat(_("Disks formatting failed. Exit code: %1.\nError output: %2"),
-            iret, stderr_from_proccess)
-        )
-      end
+    # @param [Boolean] value boolean Activate DIAG or not
+    def ActivateDiag(channel, value)
+      dasd = devices.by_id(channel)
+      return if !dasd || value == dasd.diag_wanted
 
-      nil
+      DeactivateDisk(dasd.id, dasd.diag_wanted)
+      ActivateDisk(dasd.id, value)
     end
 
-    # Get partitioninfo
-    # @param [String] disk string Disk to read info from
-    # @return GetPartitionInfo string The info
-    def GetPartitionInfo(disk)
-      outmap = Convert.to_map(
-        SCR.Execute(
-          path(".target.bash_output"),
-          Builtins.sformat("/sbin/fdasd -p '%1'", disk)
-        )
-      )
-
-      # if not an eckd-disk it's an fba-disk. fba-disks have only one partition
-      return Builtins.sformat("%11", disk) if Ops.get_integer(outmap, "exit", 0) != 0
-
-      out = Ops.get_string(outmap, "stdout", "")
-
-      regexp = "^[ \t]*([^ \t]+)[ \t]+([0-9]+)[ \t]+([0-9]+)[ \t]+([0-9]+)" \
-        "[ \t]+([^ \t]+)[ \t]+([^ \t]+([ \t]+[^ \t]+))*[ \t]*$"
-
-      l = Builtins.splitstring(out, "\n")
-      l = Builtins.filter(l) { |s| Builtins.regexpmatch(s, regexp) }
-      l = Builtins.maplist(l) do |s|
-        tokens = Builtins.regexptokenize(s, regexp)
-        Builtins.sformat(
-          "%1 (%2)",
-          Ops.get_string(tokens, 0, ""),
-          Ops.get_string(tokens, 5, "")
-        )
-      end
-      Builtins.mergestring(l, ", ")
+    # It formats the given disks showing the progress in a separate dialog
+    #
+    # @param [S390::DasdsCollection] collection of dasds to be be formatted
+    def FormatDisks(disks_list)
+      format_dialog_for(disks_list).run
     end
 
     publish variable: :devices, type: "map <integer, map <string, any>>"
```

Comment:
I'm not too familiar with those `publish` lines. But I have doubts about if it still being true after all of those changes, specially the one done at https://github.com/yast/yast-s390/pull/93/files#diff-10db1821b2b97379f01ee490a1cd79f7b24006e36cc77c26bfe802ea477e33e9R42

Reply:
If we do not replace it with **attr_accessor** then yes, it is still used